### DISCUSSION TITLE
1주차 네트워크 스터디

### DISCUSSION THREAD

**Discussion by eatingbug** [Team-Psyduck/tech-sharing#42] [Network]
### 1주차
> 2023\. 03. 20. (월) - 2023. 03. 24. (금)

컴퓨터 네트워크 기본 1
`컴퓨터 네트워크 및 인터넷 역사`

컴퓨터 네트워크 기본 2
`컴퓨터 네트워크 및 인터넷 역사`

애플리케이션 계층 1
`사용자 계층, 네트워크 애플리케이션 종류`

애플리케이션 계층 2
`Socket Programming`

전송 계층 1
`전송 계층의 개념`

**Comment by eatingbug** [comment]
컴퓨터 네트워크 기본 1
`컴퓨터 네트워크 및 인터넷 역사`

**Reply by eatingbug** [reply]
Q. 라우터 Queue 에 패킷이 가득차서 패킷 유실이 발생할 때, TCP 프로토콜이 어떻게 유실을 확인하고 재전송해주나요?

A. TCP는 패킷 유실을 확인하고 재전송하기 위해 Sequence Number, 확인 응답(ACK), 그리고 타이머를 사용한다. Sequence Number는 패킷 순서를 관리하고, ACK는 수신자가 패킷을 받았음을 송신자에게 알리며, 타이머는 ACK가 제시간에 도착하지 않을 경우 패킷을 재전송합니다. 이 메커니즘들을 통해 TCP는 신뢰성 있는 데이터 전송을 보장한다.

**Comment by eatingbug** [comment]
컴퓨터 네트워크 기본 2
`컴퓨터 네트워크 및 인터넷 역사`

**Reply by eatingbug** [reply]
Q. HTTP 는 대부분 TCP 를 기반으로 통신하는데, TCP 는 stateful 하고 HTTP 는 stateless 인 이유가 무엇인가요?

**Comment by eatingbug** [comment]
애플리케이션 계층 1
`사용자 계층, 네트워크 애플리케이션 종류`

**Reply by kkg5** [reply]
Q. Socket Functions 중 listen()은 어떤 역할을 하나요?

**Comment by eatingbug** [comment]
애플리케이션 계층 2
`Socket Programming`

**Reply by eatingbug** [reply]
Q. TCP sequence number 는 32 bit 가 할당되는데 시퀀스 번호는 4GB 이하의 데이터만 나타낼 수 있다. 데이터의 값이 2<sup>32</sup> 가 넘어가면 어떻게 되는가?

**Comment by eatingbug** [comment]
전송 계층 1
`전송 계층의 개념`
